### PR TITLE
Fix validation

### LINKED ISSUE #1376: Block validation (edit mode) doesn't work properly in page composer
#### Sonata packages

<details><summary>show</summary>
<p>

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.105.1 4.0.1  The missing Symfony Admin Generator
sonata-project/block-bundle              3.23.2  4.7.0  Symfony SonataBlockBundle
sonata-project/cache                     2.2.0   2.2.0  Cache library
sonata-project/cache-bundle              3.3.0   3.3.0  This bundle provides caching services
sonata-project/classification-bundle     3.17.0  3.17.0 Symfony SonataClassificationBundle
sonata-project/datagrid-bundle           3.4.0   3.4.0  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.14.0  1.14.0 Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.35.0  4.0.0  Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/exporter                  2.8.0   2.8.0  Lightweight Exporter library
sonata-project/form-extensions           1.12.0  1.12.0 Symfony form extensions
sonata-project/formatter-bundle          4.6.1   4.6.1  Symfony SonataFormatterBundle
sonata-project/intl-bundle               2.10.2  2.10.2 Symfony SonataIntlBundle
sonata-project/media-bundle              3.35.1  3.35.1 Symfony SonataMediaBundle
sonata-project/notification-bundle       3.13.0  3.13.0 Symfony SonataNotificationBundle
sonata-project/page-bundle               3.23.0  3.23.0 This bundle provides a Site and Page management through container and block services
sonata-project/seo-bundle                2.14.0  2.14.0 Symfony SonataSeoBundle
sonata-project/translation-bundle        2.9.1   2.9.1  SonataTranslationBundle
sonata-project/twig-extensions           1.9.0   1.9.0  Sonata twig extensions
sonata-project/user-bundle               4.12.0  4.12.0 Symfony SonataUserBundle
```

</p>
</details>

#### Symfony packages

<details><summary>show</summary>
<p>

```
$ composer show --latest 'symfony/*'
symfony/apache-pack                v1.0.1  v1.0.1  A pack for Apache support in Symfony
symfony/asset                      v4.4.27 v4.4.27 Manages URL generation and versioning of web assets such as CSS stylesheets, JavaScript...
symfony/browser-kit                v4.4.27 v4.4.27 Simulates the behavior of a web browser, allowing you to make requests, click on links ...
symfony/cache                      v4.4.31 v4.4.31 Provides an extended PSR-6, PSR-16 (and tags) implementation
symfony/cache-contracts            v2.4.0  v2.4.0  Generic abstractions related to caching
symfony/config                     v4.4.30 v4.4.30 Helps you find, load, combine, autofill and validate configuration values of any kind
symfony/console                    v4.4.30 v4.4.30 Eases the creation of beautiful and testable command line interfaces
symfony/css-selector               v4.4.27 v4.4.27 Converts CSS selectors to XPath expressions
symfony/debug                      v4.4.31 v4.4.31 Provides tools to ease debugging PHP code
symfony/debug-bundle               v4.4.27 v4.4.27 Provides a tight integration of the Symfony Debug component into the Symfony full-stack...
symfony/dependency-injection       v4.4.31 v4.4.31 Allows you to standardize and centralize the way objects are constructed in your applic...
symfony/deprecation-contracts      v2.4.0  v2.4.0  A generic function and convention to trigger deprecation notices
symfony/doctrine-bridge            v4.4.31 v4.4.31 Provides integration for Doctrine with various Symfony components
symfony/dom-crawler                v4.4.30 v4.4.30 Eases DOM navigation for HTML and XML documents
symfony/dotenv                     v4.4.29 v4.4.29 Registers environment variables from a .env file
symfony/error-handler              v4.4.30 v4.4.30 Provides tools to manage errors and ease debugging PHP code
symfony/event-dispatcher           v4.4.30 v4.4.30 Provides tools that allow your application components to communicate with each other by...
symfony/event-dispatcher-contracts v1.1.9  v2.4.0  Generic abstractions related to dispatching event
symfony/expression-language        v4.4.30 v4.4.30 Provides an engine that can compile and evaluate expressions
symfony/filesystem                 v4.4.27 v4.4.27 Provides basic utilities for the filesystem
symfony/finder                     v4.4.30 v4.4.30 Finds files and directories via an intuitive fluent interface
symfony/flex                       v1.17.1 v1.17.2 Composer plugin for Symfony
symfony/form                       v4.4.31 v4.4.31 Allows to easily create, process and reuse HTML forms
symfony/framework-bundle           v4.4.31 v4.4.31 Provides a tight integration between Symfony components and the Symfony full-stack fram...
symfony/http-client                v4.4.31 v4.4.31 Provides powerful methods to fetch HTTP resources synchronously or asynchronously
symfony/http-client-contracts      v2.4.0  v2.4.0  Generic abstractions related to HTTP clients
symfony/http-foundation            v4.4.30 v4.4.30 Defines an object-oriented layer for the HTTP specification
symfony/http-kernel                v4.4.32 v4.4.32 Provides a structured process for converting a Request into a Response
symfony/inflector                  v4.4.27 v4.4.27 Converts words between their singular and plural forms (English only)
Package symfony/inflector is abandoned, you should avoid using it. Use use `EnglishInflector` from the String component instead instead.
symfony/intl                       v4.4.31 v4.4.31 Provides a PHP replacement layer for the C intl extension that includes additional data...
symfony/mailer                     v4.4.27 v4.4.27 Helps sending emails
symfony/maker-bundle               v1.34.0 v1.34.1 Symfony Maker helps you create empty commands, controllers, form classes, tests and mor...
symfony/mime                       v4.4.31 v4.4.31 Allows manipulating MIME messages
symfony/monolog-bridge             v4.4.27 v4.4.27 Provides integration for Monolog with various Symfony components
symfony/monolog-bundle             v3.7.0  v3.7.0  Symfony MonologBundle
symfony/options-resolver           v4.4.30 v4.4.30 Provides an improved replacement for the array_replace PHP function
symfony/phpunit-bridge             v5.3.8  v5.3.8  Provides utilities for PHPUnit, especially user deprecation notices management
symfony/polyfill-intl-grapheme     v1.23.1 v1.23.1 Symfony polyfill for intl's grapheme_* functions
symfony/polyfill-intl-icu          v1.23.0 v1.23.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn          v1.23.0 v1.23.0 Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-intl-normalizer   v1.23.0 v1.23.0 Symfony polyfill for intl's Normalizer class and related functions
symfony/polyfill-mbstring          v1.23.1 v1.23.1 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72             v1.23.0 v1.23.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73             v1.23.0 v1.23.0 Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/polyfill-php80             v1.23.1 v1.23.1 Symfony polyfill backporting some PHP 8.0+ features to lower PHP versions
symfony/polyfill-php81             v1.23.0 v1.23.0 Symfony polyfill backporting some PHP 8.1+ features to lower PHP versions
symfony/process                    v4.4.30 v4.4.30 Executes commands in sub-processes
symfony/property-access            v4.4.30 v4.4.30 Provides functions to read and write from/to an object or array using a simple string n...
symfony/property-info              v4.4.31 v4.4.31 Extracts information about PHP class' properties using metadata of popular sources
symfony/proxy-manager-bridge       v4.4.27 v4.4.27 Provides integration for ProxyManager with various Symfony components
symfony/routing                    v4.4.30 v4.4.30 Maps an HTTP request to a set of configuration variables
symfony/security-acl               v3.2.0  v3.2.0  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle            v4.4.27 v4.4.27 Provides a tight integration of the Security component into the Symfony full-stack fram...
symfony/security-core              v4.4.31 v4.4.31 Symfony Security Component - Core Library
symfony/security-csrf              v4.4.27 v4.4.27 Symfony Security Component - CSRF Library
symfony/security-guard             v4.4.27 v4.4.27 Symfony Security Component - Guard
symfony/security-http              v4.4.30 v4.4.30 Symfony Security Component - HTTP Integration
symfony/serializer                 v4.4.31 v4.4.31 Handles serializing and deserializing data structures, including object graphs, into ar...
symfony/service-contracts          v2.4.0  v2.4.0  Generic abstractions related to writing services
symfony/stopwatch                  v4.4.27 v4.4.27 Provides a way to profile code
symfony/string                     v5.3.7  v5.3.7  Provides an object-oriented API to strings and deals with bytes, UTF-8 code points and ...
symfony/swiftmailer-bundle         v3.5.2  v3.5.2  Symfony SwiftmailerBundle
symfony/templating                 v4.4.30 v4.4.30 Provides all the tools needed to build any kind of template system
symfony/translation                v4.4.32 v4.4.32 Provides tools to internationalize your application
symfony/translation-contracts      v2.4.0  v2.4.0  Generic abstractions related to translation
symfony/twig-bridge                v4.4.27 v4.4.27 Provides integration for Twig with various Symfony components
symfony/twig-bundle                v4.4.30 v4.4.30 Provides a tight integration of Twig into the Symfony full-stack framework
symfony/validator                  v4.4.31 v4.4.31 Provides tools to validate values
symfony/var-dumper                 v4.4.31 v4.4.31 Provides mechanisms for walking through any arbitrary PHP variable
symfony/var-exporter               v4.4.31 v4.4.31 Allows exporting any serializable PHP data structure to plain PHP code
symfony/web-link                   v4.4.27 v4.4.27 Manages links between resources
symfony/web-profiler-bundle        v4.4.31 v4.4.31 Provides a development tool that gives detailed information about the execution of any ...
symfony/webpack-encore-bundle      v1.12.0 v1.12.0 Integration with your Symfony app & Webpack Encore!
symfony/yaml                       v4.4.29 v4.4.29 Loads and dumps YAML files
```

</p>
</details>

#### PHP version

```
$ php -v
PHP 7.4.19 (cli) (built: May  4 2021 14:24:16) ( NTS Visual C++ 2017 x64 )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.19, Copyright (c), by Zend Technologies
```

## Subject

Related to #1318 fixed in create mode by #1323. In edit mode, the problem still persist:

oIlSUQjKqJ

Testing block:

```php
<?php

declare(strict_types=1);

/*
 * This file is part of the app package.
 *
 * (c) Marco Polichetti <sviluppatore@iosystems.it>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace App\Block;

use Sonata\BlockBundle\Block\BlockContextInterface;
use Sonata\BlockBundle\Block\Service\AbstractBlockService;
use Sonata\BlockBundle\Block\Service\EditableBlockService;
use Sonata\BlockBundle\Form\Mapper\FormMapper;
use Sonata\BlockBundle\Meta\Metadata;
use Sonata\BlockBundle\Meta\MetadataInterface;
use Sonata\BlockBundle\Model\BlockInterface;
use Sonata\Form\Type\ImmutableArrayType;
use Sonata\Form\Validator\ErrorElement;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\OptionsResolver\OptionsResolver;
use Symfony\Component\Templating\EngineInterface;
use Symfony\Component\Validator\Constraints\NotBlank;

class HeaderBlockService extends AbstractBlockService implements EditableBlockService
{
    public function __construct(EngineInterface $templating)
    {
        parent::__construct($templating);
    }

    public function configureSettings(OptionsResolver $resolver)
    {
        $resolver
            ->setDefaults([
                'title' => null,
                'template' => 'block/block_title.html.twig',
            ])
        ;
    }

    public function configureCreateForm(FormMapper $form, BlockInterface $block)
    {
        $this->configureEditForm($form, $block);
    }

    public function configureEditForm(FormMapper $form, BlockInterface $block)
    {
        $form
            ->add('settings', ImmutableArrayType::class, [
                'keys' => [
                    ['title', null, [
                        'label' => 'Title',
                        'constraints' => [
                            new NotBlank(),
                        ],
                    ]]
                ],
            ])
        ;
    }

    public function prePersist(BlockInterface $block)
    {
    }

    public function preUpdate(BlockInterface $block)
    {
    }

    public function validate(ErrorElement $errorElement, BlockInterface $block)
    {
    }

    public function execute(BlockContextInterface $blockContext, Response $response = null)
    {
        return $this->renderResponse($blockContext->getTemplate(), [
            'block_context' => $blockContext,
            'block' => $blockContext->getBlock(),
            'settings' => $blockContext->getSettings(),
        ], $response);
    }

    public function getMetadata(): MetadataInterface
    {
        return new Metadata($this->getName(), null, null, null, [
            'class' => 'fa fa-header',
        ]);
    }

    public function getName()
    {
        return 'header';
    }
}
```

Can be fixed here https://github.com/sonata-project/SonataPageBundle/blob/3.x/assets/src/js/composer.js#L485 adding:

```js
headers: {
  Accept: 'text/html, application/xhtml+xml;'
},
```

@VincentLanglet can you make the same fix also this time?

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

Similar to https://github.com/sonata-project/SonataPageBundle/pull/1323

I am targeting this branch, because bugfix.

Closes #1376.

## Changelog

```markdown
### Fixed
- Block validation in edit mode
```

### CODE REVIEW

File: src/Resources/public/sonata-page.back.js [hunk 1]
```diff
@@ -491,6 +491,9 @@
                     url:     url,
                     data:    $form.serialize(),
                     type:    method,
+                    headers: {
+                        Accept: 'text/html, application/xhtml+xml;'
+                    },
                     success: function (resp) {
                         $loader.hide();
                         if (resp.result && resp.result === 'ok') {
```
Comment (phansys) [inline src/Resources/public/sonata-page.back.js:499]: If the accepted type is only HTML, how is this condition supposed to work?

Comment (VincentLanglet) [reply src/Resources/public/sonata-page.back.js:499]: Honestly, I know nothing at all about this.

The original issue was 
https://github.com/sonata-project/SonataPageBundle/issues/1318
a similar is now for edit
https://github.com/sonata-project/SonataPageBundle/issues/1376

I just did the PR as asked by @gremo

Comment (gremo) [reply src/Resources/public/sonata-page.back.js:499]: @phansys @VincentLanglet the explanation of the fix was in the original issue https://github.com/sonata-project/SonataPageBundle/issues/1318#issuecomment-855810425 if you have some question just ask

**Comment by VincentLanglet** [comment]
Please review @sonata-project/contributors

**Review by core23 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thanks @VincentLanglet

**Pull request merged**